File: exercises/advent2021/src/day1/part1.clj
(ns day1.part1
  (:require
   [clojure.string :as s]
   [day1.calc :as calc]))

(def input-file
  (str "data/day1.txt"))

(def input
  (->> input-file
       (slurp)
       (s/split-lines)
       (map #(Integer/parseInt %))))

(def result
  (calc/solve input))

(println result)

File: exercises/advent2021/src/day10/part1.py
def parse(filename: str):
    with open(filename) as file:
        return file.read().split('\n')

def solve(input: list[str]) -> int:
    matchingRules = {
        '(': ')',
        '[': ']',
        '{': '}',
        '<': '>'
    }
    points = {
        ')': 3,
        ']': 57,
        '}': 1197,
        '>': 25137
    }
    score = 0

    for chunk in input:
        brackets = []
        for _, newBracket in enumerate(chunk):
            if newBracket in matchingRules:
                brackets.append(newBracket)
            else:
                openBracket = brackets.pop()
                if matchingRules[openBracket] != newBracket:
                    score += points[newBracket]
                    break
        
    return score

def main():
    input = parse('data/day10.txt')
    result = solve(input)
    print(result)

if __name__ == '__main__':
    main()